libraries/reconstruction_cuda/cuda_gpu_multidim_array.cu

```diff
@@ -102,4 +102,189 @@ T interpolatedElementBSpline2D_Degree3(T x, T y, int xdim, int ydim, T* data)
 
 #undef LOOKUP_TABLE_LEN
 
+#define LOOKUP_TABLE_LEN 6
+
+/*
+* This version doesn't subtract shift from position in float and doesn't lose
+* precision with higher values of `x` and `y`
+*/
+
+template<typename T>
+__device__
+T interpolatedElementBSpline2D_Degree3New(int x, int y, T x_shift, T y_shift, int xdim, int ydim, T* data)
```

Comment:
I would prefer to remove the suffix' (i.e. withouth 'New'), and suffix the original one with 'Ref' (or sth. similar).
Other possibility would be to suffix it if 'Optimized' (or some variation).
What do you think?

Reply:
Ok, so, when you feel like, please prepare a new PR, where you will:

- remove the original implementation
- rename this one with 'Ref' suffix